From: Jan Blunck <jblunck@suse.de>
Subject: ARM: move mach-types.h to arch/include/asm

Although all other architecture specific includes where already moved to
arch/include/asm the mach-types.h header was still generated in the old
location.

Signed-off-by: Jan Blunck <jblunck@suse.de>
---
 arch/arm/Makefile       |    4 ++--
 arch/arm/tools/Makefile |    2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

--- a/arch/arm/Makefile
+++ b/arch/arm/Makefile
@@ -247,7 +247,7 @@ archprepare: maketools
 
 PHONY += maketools FORCE
 maketools: include/linux/version.h FORCE
-	$(Q)$(MAKE) $(build)=arch/arm/tools include/asm-arm/mach-types.h
+	$(Q)$(MAKE) $(build)=arch/arm/tools arch/arm/include/asm/mach-types.h
 
 # Convert bzImage to zImage
 bzImage: zImage
@@ -258,7 +258,7 @@ zImage Image xipImage bootpImage uImage:
 zinstall install: vmlinux
 	$(Q)$(MAKE) $(build)=$(boot) MACHINE=$(MACHINE) $@
 
-CLEAN_FILES += include/asm-arm/mach-types.h \
+CLEAN_FILES += arch/arm/include/asm/mach-types.h \
 	       include/asm-arm/arch include/asm-arm/.arch
 
 # We use MRPROPER_FILES and CLEAN_FILES now
--- a/arch/arm/tools/Makefile
+++ b/arch/arm/tools/Makefile
@@ -4,7 +4,7 @@
 # Copyright (C) 2001 Russell King
 #
 
-include/asm-arm/mach-types.h: $(src)/gen-mach-types $(src)/mach-types
+arch/arm/include/asm/mach-types.h: $(src)/gen-mach-types $(src)/mach-types
 	@echo '  Generating $@'
 	@mkdir -p $(dir $@)
 	$(Q)$(AWK) -f $^ > $@ || { rm -f $@; /bin/false; }
